src/EditorFeatures/VisualBasic/InheritanceMargin/VisualBasicInheritanceMarginService.vb

```diff
@@ -0,0 +1,109 @@
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.InheritanceMarginService
+
+    <ExportLanguageService(GetType(IInheritanceMarginService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicInheritanceMarginService
+        Inherits AbstractInheritanceMarginService
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+        End Sub
+
+        Protected Overrides Function GetMembers(root As SyntaxNode, spanToSearch As TextSpan) As ImmutableArray(Of SyntaxNode)
+            Dim typeBlockNodes = root.DescendantNodes(spanToSearch).OfType(Of TypeBlockSyntax).ToImmutableArray()
+
+            Dim builder As ArrayBuilder(Of SyntaxNode) = Nothing
+            Using ArrayBuilder(Of SyntaxNode).GetInstance(builder)
+                builder.AddRange(typeBlockNodes.Select(Function(node) node.BlockStatement))
+
+                For Each node In typeBlockNodes
+                    For Each member In node.Members
+                        'Sub and Function
+                        Dim methodBlockNode = TryCast(member, MethodBlockSyntax)
+                        If methodBlockNode IsNot Nothing Then
+                            builder.Add(methodBlockNode.BlockStatement)
+                        End If
+
+                        Dim methodStatementNode = TryCast(member, MethodStatementSyntax)
+                        If methodStatementNode IsNot Nothing Then
+                            builder.Add(methodStatementNode)
+                        End If
+
+                        'Property
+                        Dim propertyBlockNode = TryCast(member, PropertyBlockSyntax)
+                        If propertyBlockNode IsNot Nothing Then
+                            builder.Add(propertyBlockNode.PropertyStatement)
+                        End If
+
+                        Dim propertyStatementNode = TryCast(member, PropertyStatementSyntax)
+                        If propertyStatementNode IsNot Nothing Then
+                            builder.Add(propertyStatementNode)
+                        End If
+
+                        'Custom event
+                        Dim eventDeclarationNode = TryCast(member, EventBlockSyntax)
+                        If eventDeclarationNode IsNot Nothing Then
+                            builder.Add(eventDeclarationNode.EventStatement)
+                        End If
+
+                        'One line event
+                        If TypeOf member Is EventStatementSyntax Then
+                            builder.Add(member)
+                        End If
+                    Next
+                Next
+
+                Return builder.ToImmutable()
+            End Using
+        End Function
+
+        Protected Overrides Function GetIdentifierLineNumber(sourceText As SourceText, declarationNode As SyntaxNode) As Integer
```

Comment:
same feedback as above.  this is much cleaner if you just return the identifier.

however, looking at thsi now, the cleanest way would be to just have GetMembers return an array of tuples of (node, idedntifier-token).